mp/src/game/client/cdll_client_int.cpp

```diff
@@ -1273,6 +1284,64 @@ void CHLClient::PostInit()
 	}
 
 	NeoToggleConsoleEnforce();
+
+	// Detect OBS - Only on startup
+	if (neo_cl_streamermode_autodetect_obs.GetBool())
+	{
+		g_bOBSDetected = false;
+#ifdef LINUX
+		FileFindHandle_t findHdl;
+		for (const char *pszFilename = filesystem->FindFirst("/proc/*", &findHdl);
+			 pszFilename && !g_bOBSDetected;
+			 pszFilename = filesystem->FindNext(findHdl))
+		{
+			if (!filesystem->FindIsDirectory(findHdl))
+			{
+				continue;
+			}
+
+			char szCmdlinePath[256];
+			V_sprintf_safe(szCmdlinePath, "/proc/%s/cmdline", pszFilename);
+			if (!filesystem->FileExists(szCmdlinePath))
+			{
+				continue;
+			}
+
+			// NEO NOTE (nullsystem): At this point, cannot just do filesystem->ReadFile
+			// as it won't output anything. Instead just cat the file and fetch the line.
+			char szCmdCat[256 + 16];
+			V_sprintf_safe(szCmdCat, "cat \"%s\"", szCmdlinePath);
+			FILE *fp = popen(szCmdCat, "r");
+			if (fp)
+			{
+				char szCmdline[512] = {};
+				if (fgets(szCmdline, sizeof(szCmdline), fp))
+				{
+					// NEO TODO (nullsystem): Check on OBS flatpak, snaps, nix, and appimage.
+					// Could they give out different cmdline?
+					g_bOBSDetected = (V_strcmp(szCmdline, "obs") == 0);
+				}
+				pclose(fp);
+			}
+		}
+#else
+		HANDLE hSnapShot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+		PROCESSENTRY32 processInfo = {};
+		processInfo.dwSize = sizeof(PROCESSENTRY32);
```

Comment:
Would be prudent to check the `CreateToolhelp32Snapshot` retval against the `INVALID_HANDLE_VALUE` error value, as the MS docs for `CloseHandle` seem to imply that attempting to close an invalid handle would throw, at least in debug mode.